File: JavaScript/valid-parentheses.js
// 20. Valid Parentheses

var isValid = function(s) {
    if (s.length % 2 !== 0) return false

    const chars = {
        '}': '{',
        ')': '(',
        ']': '['
    }
    const stack = []

    for (let i = 0; i < s.length; i++) {
        if (chars[s[i]]) {
            if (chars[s[i]] === stack[stack.length - 1]) {
                stack.pop()
            } else {
                return false
            }
        } else {
            stack.push(s[i])
        }
    }

    return !stack.length 
};

File: Python/Valid_Parentheses.py
# 20. Valid Parentheses

class Solution:
    def isValid(self, s: str) -> bool:
        pare = {
            '(': ')',
            '[': ']',
            '{': '}'
        }
        stack = []
        for ch in s:
            if ch in pare.keys():
                stack.append(ch)
            elif len(stack) == 0 or ch != pare[stack.pop()]:
                return False
        return len(stack) == 0
